packages/cli/src/commands/server/runServer.ts

```diff
@@ -20,22 +20,14 @@ import loadMetroConfig from '../../tools/loadMetroConfig';
 import releaseChecker from '../../tools/releaseChecker';
 
 export type Args = {
-  assetPlugins?: string[];
   cert?: string;
   customLogReporterPath?: string;
   host?: string;
   https?: boolean;
-  maxWorkers?: number;
   key?: string;
-  platforms?: string[];
-  port?: number;
```

Comment:
A bit controversial, but `--port` can be set in the configuration of Metro and I don't think it's something users would like to change on per run basis. If it's non-standard, they would probably prefer to persist it inside a custom configuration.

Reply:
Let's leave it as is. `port` is important and I use it frequently. I'd use it more often, if iOS worked with it (Android doesn't have issues) – we need to fix that separately

Reply:
Is this something you change as you run it? E.g. is the `--port` set "always" to same value or you change it each time you run the command?

I am wondering whether being able to set port in `metro.config.js` (just like with Webpack) wouldn't be enough.

Reply:
The typical scenario is having multiple RN apps and being able to run them on different Metro instances. Passing `--port` is convenient to do from the command line. I'd see removing this as controversial, especially that it doesn't bring too much complexity

Reply:
> The typical scenario is having multiple RN apps and being able to run them on different Metro instances.

Do you mean having multiple separate React Native projects running at the same time? Wouldn't the ability to set `port: XXX` in their respective configuration files be equally easy? 

All I am saying is that if you already change a port, you may have a reason to do it. And in that case, you may prefer to store it permanently in the configuration file and forget about it.

Reply:
I am not really opinionated here - just wanted to kick start the discussion and challenge how do we use this while running our apps these days.

Before finally deciding on leaving it as is, I'd like to get some extra thoughts on that from the community.

Reply:
I may work on N projects with N different teams – for each of which their project is the only one they work with. How would I justify changing configuration permanently for my own workflow to them? What if one of them start to have similar setup, but want different ports? With `--port` flag I don't have to change any files myself all the time and still keep the workflow.

I understand your urge to remove as much code as possible, but specifying a port is universal across many different bundlers, so it doesn't lock us in with Metro or any other.

Reply:
Maybe it's better to leave `port` in CLI option as well.

Personally I think having ability to specify separate config files is more useful (e.g. `--config metro.prod.config.js`). Not sure if CLI supports it.

Reply:
Yeah, CLI supports that @satya164 and that was the reason why I considered `--port` redundant (an unnecessary complication of configuration layers).

Reply:
> I understand your urge to remove as much code as possible,

@thymikee, it's not the urge to remove code. I am simply challenging the use cases and configurations that we have established 3 to 4 years ago, while not having some of the current Metro options and features.

The intent here is to not make the UX more complex or to remove the code "for the sake of it", but to check whether there are better alternatives right now and how do they compare against the preferred developer workflow.

Reply:
> I may work on N projects with N different teams – for each of which their project is the only one they work with. How would I justify changing configuration permanently for my own workflow to them? What if one of them start to have similar setup, but want different ports? With --port flag I don't have to change any files myself all the time and still keep the workflow.

This makes sense. It's still probably a very edge-case to have multiple React Native apps running at the same time, but we can leave it as-is for now and revisit in the future when Metro configuration establishes its presence a bit more.

Reply:
I agree this is edge-case. I don't bother too much personally, but I know people feel strongly about their workflows 🤷‍♂. It may make sense to start a discussion on that in a new GH issue and share it on Twitter or somewhere to see how users feel about it.